File: src/main/java/site/liangbai/realhomehunt/listener/player/ListenerPlayerMove.java
/*
 * RealHomeHunt
 * Copyright (C) 2021  Liangbai
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published
 * by the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */

package site.liangbai.realhomehunt.listener.player;

import org.bukkit.Location;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.player.PlayerMoveEvent;
import site.liangbai.lrainylib.annotation.Plugin;
import site.liangbai.realhomehunt.util.Players;

@Plugin.EventSubscriber
public final class ListenerPlayerMove implements Listener {
    @EventHandler
    public void onPlayerMove(PlayerMoveEvent event) {
        Location loc1 = event.getFrom();

        Location loc2 = event.getTo();

        if (loc1.getX() != loc2.getX() || loc1.getZ() != loc2.getZ()) {
            Players.push(event.getPlayer());
        }
    }
}

File: RealHomeHunt/src/main/java/site/liangbai/realhomehunt/internal/listener/player/ListenerPlayerMove.kt
/*
 * RealHomeHunt
 * Copyright (C) 2022  Liangbai
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published
 * by the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */

package site.liangbai.realhomehunt.internal.listener.player

import org.bukkit.event.player.PlayerMoveEvent
import site.liangbai.realhomehunt.util.Players
import taboolib.common.platform.event.SubscribeEvent

internal object ListenerPlayerMove {
    @SubscribeEvent
    fun onPlayerMove(event: PlayerMoveEvent) {
        val loc1 = event.from
        val loc2 = event.to
        if (loc1.x != loc2!!.x || loc1.z != loc2.z) {
            Players.push(event.player)
        }
    }
}